src/edu/washington/escience/myria/operator/Split.java

```diff
@@ -0,0 +1,146 @@
+          String[] splits = pattern.split(colValue, -1);
+          for (String segment : splits) {
+            splitColBuilder.appendString(segment);
+            // For each split segment, duplicate the values of all other columns in this row.
+            for (int inputColIdx = 0; inputColIdx < inputColIndexToOutputColIndex.length; ++inputColIdx) {
+              int outputColIdx = inputColIndexToOutputColIndex[inputColIdx];
+              if (outputColIdx != -1) {
+                TupleUtils.copyValue(tb.asColumn(inputColIdx), rowIdx, colBuilders.get(outputColIdx));
+              }
+            }
+          }
+        } catch (Exception e) {
+          throw new DbException(e);
+        }
+      }
+      // TODO: this should be factored out into a common utility method
+      List<Column<?>> columns = new ArrayList<>();
+      for (ColumnBuilder<?> colBuilder : colBuilders) {
+        columns.add(colBuilder.build());
+      }
+      return new TupleBatch(outputSchema, columns);
+    }
+    return null;
+  }
+
+  @Override
+  protected void init(final ImmutableMap<String, Object> execEnvVars) throws DbException {
+    final Operator child = getChild();
+    Preconditions.checkNotNull(child, "Child operator cannot be null");
+    final Schema inputSchema = child.getSchema();
+    Preconditions.checkNotNull(inputSchema, "Child schema cannot be null");
+    splitColIdx = inputSchema.columnNameToIndex(columnName);
+    final Type colType = inputSchema.getColumnType(splitColIdx);
+    Preconditions.checkState(colType == Type.STRING_TYPE, "Column to split \"" + columnName
+        + "\" must have type STRING_TYPE but has type " + colType);
+
+    // The only difference between input and output schema is that the input column being split is removed
+    // from the output schema and the (anonymous) output column containing split results appears at the
+    // final ordinal position.
+    inputColIndexToOutputColIndex = new int[inputSchema.numColumns()];
+    int[] colsToKeep = new int[inputSchema.numColumns() - 1];
+    for (int colIdx = 0; colIdx < inputSchema.numColumns(); ++colIdx) {
+      if (colIdx == splitColIdx) {
+        // Omit split column index from list of columns of output schema.
+        // -1 is a sentinel value indicating that the split column results appear in the last column of the output
+        // schema.
+        inputColIndexToOutputColIndex[colIdx] = -1;
+      } else if (colIdx > splitColIdx) {
+        // Offset output column index to compensate for deleted split column index.
+        inputColIndexToOutputColIndex[colIdx] = colIdx - 1;
+        colsToKeep[colIdx - 1] = colIdx;
+      } else {
+        inputColIndexToOutputColIndex[colIdx] = colIdx;
+        colsToKeep[colIdx] = colIdx;
+      }
+    }
+    final Schema subSchema = inputSchema.getSubSchema(colsToKeep);
+    outputSchema = Schema.appendColumn(subSchema, Type.STRING_TYPE, "_splits");
```

Comment:
Right, I put it in init() because I assumed each call to generateSchema() (wherever that happens in the framework) should return the same schema (and therefore should be memoized), and also because I happened to need outputSchema in fetchNextReady(), and couldn't be sure that generateSchema() would have been called by that point. If I can assume that, then I could move the schema-generating logic to generateSchema, but I would think I still want to cache the result in an instance field.